internal/install/static_snapshot_yml.go

```diff
@@ -74,25 +74,41 @@ services:
       package-registry:
         condition: service_healthy
 
-  elastic-agent:
+  fleet-server:
     image: ${ELASTIC_AGENT_IMAGE_REF}
     depends_on:
       elasticsearch:
         condition: service_healthy
       kibana:
         condition: service_healthy
+    healthcheck:
+      test: "curl -f http://127.0.0.1:8220/api/status | grep HEALTHY 2>&1 >/dev/null"
+      retries: 60
+      interval: 1s
+    hostname: docker-fleet-server
+    environment:
+    - "FLEET_SERVER_ENABLE=1"
+    - "FLEET_SERVER_INSECURE_HTTP=1"
+    - "KIBANA_FLEET_SETUP=1"
+    - "KIBANA_FLEET_HOST=http://kibana:5601"
+    - "FLEET_SERVER_HOST=0.0.0.0"
+    ports:
+      - "127.0.0.1:8220:8220"
+
+  elastic-agent:
+    image: ${ELASTIC_AGENT_IMAGE_REF}
+    depends_on:
+      fleet-server:
+        condition: service_healthy
     healthcheck:
       test: "sh -c 'grep \"Agent is starting\" /usr/share/elastic-agent/elastic-agent.log*'"
-      retries: 30
+      retries: 90
       interval: 1s
     hostname: docker-fleet-agent
     environment:
     - "FLEET_ENROLL=1"
-    - "FLEET_ENROLL_INSECURE=1"
     - "FLEET_INSECURE=1"
-    - "FLEET_SETUP=1"
-    - "FLEET_URL=http://kibana:5601"
-    - "KIBANA_HOST=http://kibana:5601"
+    - "FLEET_URL=http://fleet-server:8220"
```

Comment:
I wonder how in this scenario the elastic-agent will get the right enrollment token. We might still have to read it from Kibana.

Reply:
It's probably reading it from ES but how does the agent know what ES to call?

Reply:
Would prefer the usage of `https` here as well. But you will still need the `FLEET_INSECURE=1` so it sets `ssl.verification_mode: none`.